### PR TITLE
SONiC Yang model for sFlow feature

### PR DESCRIPTION
**Pull request by ArthiSivanantham**
Signed-off-by: ArthiSivanantham <arthi_sivanantham@dell.com>

#### Why I did it
SONiC YANG model support for sFlow feature.

#### How I did it
Defined various sFlow YANG containers and lists based on config-DB schema.

#### How to verify it
Successful build of the following packages:
make target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl
make target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog
SONiC Yang model support for sFlow feature.

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:69]: Please add description for the fields.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:47]: Please add the default value 6343 here.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:100]: Please plan to add mgmt & loopback interfaces as well whenever those YANG models are available.

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:157]: This code should be enabled once mgmt port YANG model is merged.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:52]: Addressed

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:75]: Addressed

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:70]: Default admin-status?

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:113]: I've added mgmt interface type and have commented it for now. Will enable it once the pending PR is merged, https://github.com/Azure/sonic-buildimage/pull/7252.

Regarding loopback interface, when the support for LOOPBACK table is in community, will add it here.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:89]: What's the default global status?

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:157]: Yes, waiting for PR https://github.com/Azure/sonic-buildimage/pull/7252 to be merged.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:70]: Default admin status is "up". 
https://github.com/Azure/sonic-swss/blob/master/cfgmgr/sflowmgr.cpp#L279

I've addressed the comment.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:90]: Default admin status (for global table) is "up".
https://github.com/Azure/sonic-swss/blob/master/cfgmgr/sflowmgr.cpp#L191

I've addressed the comment.

**Comment by ArthiSivanantham** [comment]
@lguohan, @praveen-li - Can you please review this PR ?

**Comment by ArthiSivanantham** [comment]
@lguohan, @praveen-li - Can you please review this PR ?

**Review by venkatmahalingam — DISMISSED** [review]

**Comment by ArthiSivanantham** [comment]
@anshuv-mfst - Can you please add the project as yang for this PR ?

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (rathnasabapathyv) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:109]: Please add portchannel as well here.

Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:115]: Addressed.

**Review by rathnasabapathyv — APPROVED** [review]

**Review by venkatmahalingam — DISMISSED** [review]

**Comment by ArthiSivanantham** [comment]
@lguohan  -  Can you please check on this failure ?

failure_logs

**Comment by anshuv-mfst** [comment]
Hi @renukamanavalan - could you please review.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:66]: How is interface all handled here?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-aaa.yang:22]: What about defaults ?

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:41]: Is our buildimage ready for these new types chap & mschap ?

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:102]: Is this supported?

Is this to route through mgmt port?

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:76]: Do we support this server level override for timeout, auth_type & passkey ?

If yes, what about src-intf override.

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:156]: If we don't need, please remove.

**Comment by ArthiSivanantham** [comment]
As suggested by @lguohan, I'm keeping this PR only for SFLOW and have a separate PR for AAA/TACACS,
https://github.com/Azure/sonic-buildimage/pull/7671

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:66]: The key will be updated to include 'all' option.

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-system-tacacs.yang]: Yes, Tacacs+ server reachability via mgmt VRF is supported, mgmt VRF can have just mgmt port or data port (in-band-mgmt).

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang]: default is local, will add default login as 'local'

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-system-tacacs.yang]: yes but server-level override for src-intf is not supported in the backend, YANG is simply capturing what's supported in the config-DB today.

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:66]: Addressed

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang]: Addressed in PR https://github.com/Azure/sonic-buildimage/pull/7671

**Comment by venkatmahalingam** [comment]
> @ArthiSivanantham - please address Sudarshan's comment, please comment out in input vlan to pass build checks.

Hello Anshu, Arthi already addressed Sudharsan's comment and I see the YANG to accept 'all' as key, @dgsudharsan please take a look and approve if no more comments, thanks.

**Review by venkatmahalingam — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:85]: The sample rate is valid only for interface "port_name" nad not for interface "all". Can we please add the dependency?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:84]: Addressed

**Comment by ArthiSivanantham** [comment]
@dgsudharsan - All comments are addressed. Can you please review and approve the PR ? Thanks.

**Review by rathnasabapathyv — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-sflow.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-sflow.yang:72]: Please add description for 'all' field.

Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-sflow.yang:72]: Addressed

**Review by venkatmahalingam — DISMISSED** [review]

**Review by venkatmahalingam — DISMISSED** [review]

**Review by dgsudharsan — DISMISSED** [review]

**Review by venkatmahalingam — DISMISSED** [review]

**Review by rathnasabapathyv — APPROVED** [review]

**Comment by bhavini-gada** [comment]
@dgsudharsan - Can you please review/approve ?

**Comment by bhavini-gada** [comment]
@lguohan - Can you please merge this one ?

**Review by dgsudharsan — APPROVED** [review]

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json:7]: plz fix alignment

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json:219]: If agent ID is not mandatory for this test case we can remove it, keep shorter config.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json
Comment (ArthiSivanantham) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json:7]: Addressed

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json
Comment (ArthiSivanantham) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/sflow.json]: Addressed

**Comment by ArthiSivanantham** [comment]
All review comments are addressed.
@praveen-li @dgsudharsan @venkatmahalingam @rathnasabapathyv - Can you all please re-approve this PR ?

**Review by dgsudharsan — APPROVED** [review]

**Review by venkatmahalingam — APPROVED** [review]

**Review by rathnasabapathyv — APPROVED** [review]

**Comment by bhavini-gada** [comment]
@lguohan - Could you please merge this PR ?

**Comment by ArthiSivanantham** [comment]
@lguohan - Can you please merge this PR ?

**Comment by bhavini-gada** [comment]
@lguohan / @prsunny - Can you please merge this PR ?

**Comment by anshuv-mfst** [comment]
Hi @renukamanavalan or @prsunny - could you please help with merge, thanks.

**Review by prsunny — APPROVED** [review]

**Pull request merged**
